File: clients/nodejs/$dotnet-test-server/UiPath.Ipc.TestServer/UiPath.Ipc.TestServer/Contract.cs
using System.Threading;
using System.Threading.Tasks;

namespace UiPath.Ipc.TestServer
{
    public static class Contract
    {
        public sealed class Complex
        {
            public double X { get; set; }
            public double Y { get; set; }
        }

        public interface IService
        {
            Task<Complex> AddAsync(Complex a, Message<Complex> b, CancellationToken ct = default);
            Task StartTimerAsync(Message message);
        }
        public interface ICallback
        {
            Task<double> AddAsync(double a, double b);
            Task<string> TimeAsync(string info);
        }
    }
}

File: clients/nodejs/projects/@uipath-ipc-demo-client/src/contract.ts
import { CancellationToken, Message, __hasCancellationToken__, __returns__ } from '@uipath/ipc';

export class Complex {
    constructor(
        public readonly X: number,
        public readonly Y: number
    ) { }
}

export class IService {
    @__hasCancellationToken__
    @__returns__(Complex)
    public AddAsync(a: Complex, b: Message<Complex>, c: CancellationToken = CancellationToken.none): Promise<Complex> { throw null; }

    public StartTimerAsync(message: Message<void>): Promise<void> { throw null; }
}
export interface ICallback {
    AddAsync(a: number, b: number): Promise<number>;
    TimeAsync(info: string): Promise<string>;
}
